### DISCUSSION TITLE
Deprecating extended keybindings

### DISCUSSION THREAD

**Discussion by numToStr** [numToStr/Comment.nvim#247] [Announcements]
Hi All,

I am thinking of deprecating _and eventually removing_ the extended keybindings from the plugin. All these keybindings are documented in the [README](https://github.com/numToStr/Comment.nvim#extended-mappings). This decision is coming because of the following reasons:

1. These are rarely used (main motivation)
2. RHS of keybindings are static and can't be configured.
3. Not documented in `:help comment-nvim` doc.
4. Maintaining keybindings is generally a pain.

> 2. and 3. can solved but doing that is just extra work without any tangible benefit.

Note that I'll only remove *keybindings*, lua API is and will always be available.

Let me know what you think about this decision :)

**Comment by gegoune** [comment]
I very often use keybindings from 'extra'. Will create my own maps if you remove them, just letting you know that they are used. :)

**Reply by IamGianluca** [reply]
Same. I also use them on a regular basis and are what made me consider switching to ‘comment.nvim’ :-)

**Reply by numToStr** [reply]
@gegoune @IamGianluca sorry about the confusion but I am thinking of removing `extended` mappings. So no need to worry about `extra` mappings :)

**Comment by numToStr** [comment]
I am so dumb. IDK how I confused `extra` with `extended` keybindings.

*To be clear, I am deprecating `extended` keybindings*

**Reply by gegoune** [reply]
Not using those. :)

**Reply by IamGianluca** [reply]
I’m not using those either 👍

**Reply by caarlos0** [reply]
Not using them either!

**Comment by proofer** [comment]
This might be a good excuse for me to ask: what is the basis for categorizing keybindings into basic, extra, and (for the time being) extended? What is the functional/operational difference between the categories?

I, too, am not using the extended ones.

**Reply by numToStr** [reply]
That's a nice question. So, I separated keybindings, from the early days of the plugin, into 3 categories: 1) basic/core: that have all the toggle comment actions, 2) extra: provides extra functionality which is common and useful for many, 3) extended: only does comment and uncomment, and were straight port from `tcommment` plugin for easy migration.

Now the problem! Maintaining keybindings is a pain and sometimes harder than the plugin itself. People have hard opinions about them, and they want to customize every piece of it which is not feasible for plugin-author. See https://github.com/numToStr/Comment.nvim/issues/182 https://github.com/numToStr/Comment.nvim/issues/119 https://github.com/numToStr/Comment.nvim/issues/117

I really want to improve the customization and extensibility of keybindings without additional overhead. But This will be a breaking change that's why I am hesitant to do it as I know, at least, thousands of people are dependent on this plugin. So, the minimum I can do is remove `extended` keybindings which are rarely used and then design customizability around rest of the keybindings.

**Reply by proofer** [reply]
That's a nice full response -- thanks!

**Comment by numToStr** [comment]
Good Riddance #254 :)
